hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -481,7 +481,10 @@ trait TableReaderWithExtraUID extends TableReader {
 abstract class TableReader {
   def pathsUsed: Seq[String]
 
-  def apply(ctx: ExecuteContext, requestedType: TableType, dropRows: Boolean): TableValue
+  def toExecuteIntermediate(ctx: ExecuteContext, requestedType: TableType, dropRows: Boolean): TableExecuteIntermediate = {
+    assert(!dropRows)
+    TableExecuteIntermediate(_lower(ctx, requestedType))
```

Comment:
I like this interface change. One thing I'd suggest is to remove the `TableExecuteIntermediate` companion object and construct `TableStageIntermediate` and `TableValueIntermediate` directly.

Reply:
That's not a bad change, but given that it will all be going away when everything is lowered, I don't think there's much benefit.